internal/commands/push.go

```diff
@@ -107,44 +108,70 @@ func NewPushCommand(ctx context.Context, logger *log.Logger) *cobra.Command {
 }
 
 func pushBundle(ctx context.Context, repository string, path string) (*ocispec.Descriptor, error) {
-	cli, err := auth.NewClient()
+	cli, err := dockerauth.NewClient()
 	if err != nil {
 		return nil, fmt.Errorf("get auth client: %w", err)
 	}
 
-	resolver, err := cli.Resolver(ctx, http.DefaultClient, false)
+	opts := []auth.ResolverOption{auth.WithResolverClient(http.DefaultClient)}
+	resolver, err := cli.ResolverWithOpts(opts...)
 	if err != nil {
 		return nil, fmt.Errorf("docker resolver: %w", err)
 	}
 
-	memoryStore := content.NewMemoryStore()
+	memoryStore := content.NewMemory()
 	layers, err := buildLayers(ctx, memoryStore, path)
 	if err != nil {
 		return nil, fmt.Errorf("building layers: %w", err)
 	}
 
-	extraOpts := []oras.PushOpt{oras.WithConfigMediaType(openPolicyAgentConfigMediaType)}
-	manifest, err := oras.Push(ctx, resolver, repository, memoryStore, layers, extraOpts...)
+	registry := content.Registry{Resolver: resolver}
+	if err != nil {
+		return nil, fmt.Errorf("creating registry: %w", err)
+	}
```

Comment:
Same comment as in `pull.go`, this error check is reusing the same error from `buildLayers()` above since the `registry` assignment does not return an error. I think this can also be removed.